avocado/plugins/list.py

```diff
@@ -165,6 +168,10 @@ def configure(self, parser):
                             'Current: %(default)s')
         loader.add_loader_options(parser)
 
+        filtering = parser.add_argument_group('filtering parameters')
+        filtering.add_argument('--filter-by-tags', metavar='TAGS',
+                               action='append')
```

Comment:
Other options are:

* `--filter-by-tags '-foo,bar'`
* `--filter-by-tags=foo,-bar`

I've tried with `!`, but it's a lot worse.  The problem with `~` is that it also has a special meaning for the shell, and is not known as a `not` operator as `-` or even `!` are.